app/components/channel_list/categories/categories.tsx

```diff
@@ -44,7 +46,12 @@ const Categories = ({categories, currentChannelId, currentUserId, currentTeamId,
 
     const renderCategory = useCallback((data: {item: CategoryModel | string[]}) => {
         if (Array.isArray(data.item)) {
-            return <UnreadCategories unreadChannels={unreadChannels}/>;
+            return (
+                <UnreadCategories
+                    currentChannelId={currentChannelId}
```

Comment:
To avoid re-renders we can let UnreadCategories to fetch itself the currentChannelId. Nevertheless, that can be covered by my performance PR.

Reply:
Yes, let's look at it in the perf PR as it will re-render itself regardless but `categories` would not re-render as is now